### ISSUE TITLE
Failed (22: Invalid argument) while sending mp4 to client

### ISSUE THREAD

**Issue by buldezir** [ShaneIsrael/fireshare#252]
**Describe the bug**
Error when directio option is enabled
```
[crit] 9#9: *3 pread() "/processed/video_links/9d0c4426399189a22097754641ea5e9f.mp4" failed (22: Invalid argument) while sending mp4 to client
```
I googled a lot about this error, and could not find anything relevant.

**Additional context**
Docker volume directory is actually mounted smb share (cifs mount)

**Comment by ShaneIsrael** [comment]
@buldezir Can you try using the `shaneisrael/fireshare:develop` docker build. If you already are, re-pull the image. I updated it and would like to see if that update solves your problem.

**Comment by Krojack** [comment]
Did anything happen with this? I noticed I'm getting this today.

**Comment by ShaneIsrael** [comment]
@Krojack  Seeing as this was 2 years ago and the guy never replied I have to assume whatever I did at the time either fixed it or he resolved the issue himself. 

I would need a lot more info on your setup and how you are accessing your videos. (i.e is your mounted volume over an smb network drive?). 

This isn't an error I've ever seen before myself (but I don't have my volumes mounted of a network) and other than you post is the only other time I've seen it mentioned. So there's not likely much I'll be able to do other than suggest not mounting of a network if that is what you are doing.

**Comment by ModestoHagney** [comment]
Just set Fireshare up today and I'm getting this error.

Uploaded a couple videos using the webui, had to remux them with FFMpeg from mkv to mp4.
They upload fine. Origonally when I try to play locally i got: `Playback Error: 2152398868: Request failed`
When I try to play remotely via caddy/cloudflare i get: `Playback Error: 502: Bad Gateway`

Now when I tried again to get docker logs i just got this: `Playback Error
NS_ERROR_NET_RESET (0x804b0014) - Download aborted`

The videos are short but reccorded in H.264 MKV with OBS, I'm gonna try again now that I've changed it to MP4. Also they're 5120x1440.

The docker error is: `fireshare  | 2026-05-15T06:27:00.879935562Z 2026/05/15 06:27:00 [crit] 37#37: *978 pread() "/processed/video_links/8d8038f8df5be7f206a41983da4c71ba.mp4" failed (22: Invalid argument) while sending mp4 to client, client: 10.0.0.50, server: _, request: "GET /_content/video/8d8038f8df5be7f206a41983da4c71ba.mp4 HTTP/1.1", host: "10.0.0.30:5005", referrer: "http://10.0.0.30:5005/"`

The error for remote connection is the same just with my domain instead of server IP.
the videos, images and processed folders are just mounted on my zfs volume mapped correctly in docker, permissions are set to my 1000:1000 user.

Edit: After some testing it seems it could be an h.264 NVENC issue with resolutions higher than 4096 in either direction, OBS wouldn't let me use NVENC. Changing to CPU encoder in OBS and uploading that works fine.
